File: doc-examples/example-toml-java/src/main/java/example/TomlService.java
package example;

import io.micronaut.serde.ObjectMapper;
import io.micronaut.serde.toml.TomlObjectMapper;
import jakarta.inject.Named;
import jakarta.inject.Singleton;

@Singleton
final class TomlService {
    private final ObjectMapper tomlMapper;

    TomlService(@Named(TomlObjectMapper.NAME) ObjectMapper tomlMapper) {
        this.tomlMapper = tomlMapper;
    }

    ObjectMapper tomlMapper() {
        return tomlMapper;
    }
}

File: doc-examples/example-toml-kotlin/src/main/kotlin/example/TomlService.kt
package example

import io.micronaut.serde.ObjectMapper
import io.micronaut.serde.toml.TomlObjectMapper
import jakarta.inject.Named
import jakarta.inject.Singleton

@Singleton
class TomlService(@Named(TomlObjectMapper.NAME) private val tomlMapper: ObjectMapper) {

    fun tomlMapper(): ObjectMapper = tomlMapper
}
